### ISSUE TITLE
Native fullscreen behaviour on macOS

### ISSUE THREAD

**Issue by arlyon** [runelite/runelite#730]
The behaviour when pressing the "full screen" button on macOS has been to occupy the entire screen instead of just resizing to fit the current screen. 

[Here is a video to illustrate what I've got working](https://gfycat.com/ArtisticGleefulKagu)

I've managed to get it working with the non-chrome menu bar (ie the "Enable custom window chrome" setting off) but wanted to see what the consensus was before adding any more platform-specific code.

**Comment by arlyon** [comment]
The needed function is in the `com.apple.eawt` library which is typically only available on macs, which means it would need to be called using reflection. Here is an example implementation:

```java
/**
 * Enables the osx native fullscreen if running on a mac.
 * @param gui The gui to enable the fullscreen on.
 */
public static void tryEnableOSXFullscreen(ClientUI gui) {
	try
	{
		Class.forName("com.apple.eawt.FullScreenUtilities")
			.getMethod("setWindowCanFullScreen", Window.class, boolean.class)
			.invoke(null, gui, true);
	} catch (ClassNotFoundException | NoSuchMethodException | IllegalAccessException | InvocationTargetException ignored)
	{
		// not running macOS, ignore
	}
}
```

**Comment by deathbeam** [comment]
Well, that should already be in PR then 👍 Maybe adding check if running on Mac (check the Notifier class for example on sort of correct checks, maybe that should be extracted to some util class anyway) instead of just trying it everytime

**Comment by Adam-** [comment]
Fixed in #732
